src/main/scala/chargepoint/docile/dsl/expectations/Ops.scala

```diff
@@ -57,6 +57,35 @@ trait Ops {
     }
   }
 
+  def expectAllIgnoringUnmatched[T](expectations: IncomingMessageProcessor[T]*)(implicit awaitTimeout: AwaitTimeout): Seq[T] = {
+
+    def loop(matchesCount: Int, results: IndexedSeq[Option[T]]): IndexedSeq[Option[T]] = {
+      val Seq(m) = awaitIncoming(1)
+      val processorIndex = expectations.indexWhere(_.accepts(m))
+
+      if (processorIndex < 0) {
+        logger.info(s"Ignoring message $m")
+        loop(matchesCount, results)
+
+      } else {
+        val p = expectations(processorIndex)
+        p.fireSideEffects(m)
+        val result = p.result(m)
+        val nextResults = results.updated(processorIndex, Some(result))
+        val nextMatchesCount: Int = results(processorIndex) match {
+          case Some(_) => matchesCount
+          case None => matchesCount + 1
+        }
+
+        logger.info(s"Received $processorIndex: $m, $nextMatchesCount to go")
+
+        loop(nextMatchesCount, nextResults)
+      }
+    }
+
+    loop(0, IndexedSeq.fill(expectations.size)(None)).flatten
```

Comment:
Does `loop` ever return something?

Reply:
Yes, the results of the commands that have been matched.